hmy/api_backend.go

```diff
@@ -443,7 +460,21 @@ func (b *APIBackend) GetValidatorInformation(
 func (b *APIBackend) GetMedianRawStakeSnapshot() (
 	*committee.CompletedEPoSRound, error,
 ) {
-	return committee.NewEPoSRound(b.hmy.BlockChain())
+	res, err := b.SingleFlightRequest(
+		"median",
```

Comment:
This doesn't work because the result keep changing every block, And the key is always “median" it will give out stale data.

Reply:
This RPC can not be cached simply like this at this high level. 
Can only go deeper and cache on lower-level computations.

Reply:
the cache is cleared below every block. why not cache the whole rpc? there are mainly three steps if we were to cache inner computations: 1) prepreOrder (preprocessing), which should be most computational, 2) followed by median which should be second most computational, and 3) post processing (lite one). Even the inner computations (1 & 2) will have to be cache block-level, which is same as above right?

Reply:
The right cache should include the block number of the current block in chain..

Reply:
I see you discarded the cache every 8 seconds. But let's do the correct way of adding block num to the key

Reply:
done.